includes/template.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**********************************************************
+* Global variables that will hold the extracted portions
+* of the HTML file
+**********************************************************/
+$headContents = "";
+$pageTitle = "";
+$bodyContents = "";
+
+/**********************************************************
+* Extracts the head and title sections, as well as the
+* body portion. Void return.
+**********************************************************/
+function extractContents($page)
+{
+    extractHeadAndTitle($page);
+    extractBody($page);
+}
+
+/**********************************************************
+* Extracts the head and title sections of the HTML file,
+* if either or both of them exist. Extracted portions are
+* stored in global variables for files that include this
+* one. Void return.
+**********************************************************/
+function extractHeadAndTitle($page)
+{
+    global $headContents;
+    global $pageTitle;
+
+    $pattern = "#<head.*?>(.*?(<title.*?>(.*?)</title.*?>).*?|.*)</head.*?>#is";
+    $matches = array();
+
+    if(preg_match($pattern, $page, $matches))
+    {
+        $headContents = trim($matches[1], "\r\n");
+        
+        if(isset($matches[3]))
+        {
+            $pageTitle = trim($matches[3]);
+            $titlePattern = "#$matches[2]#";
+
+            $headContents = preg_replace($titlePattern, '', $headContents);
```

Comment:
I don't think we want to do a `preg_replace` here, since that will cause the `<title>...</title>` tag to be interpreted as a regular expression. Maybe do [str_replace](http://php.net/manual/en/function.str-replace.php) instead.